src/sql/workbench/services/connection/browser/utils.ts

```diff
@@ -14,5 +15,22 @@ import { enableSqlAuthenticationProviderConfig, mssqlProviderName } from 'sql/pl
  * @returns True if provider is MSSQL and Sql Auth provider is enabled.
  */
 export function isMssqlAuthProviderEnabled(provider: string, configService: IConfigurationService): boolean {
-	return provider === mssqlProviderName && (configService?.getValue(enableSqlAuthenticationProviderConfig) ?? true);
+	return provider === mssqlProviderName && isMSALAuthLibraryEnabled(configService) && (configService?.getValue(enableSqlAuthenticationProviderConfig) ?? true);
```

Comment:
If you're going to check for configService being undefined you should update the parameter to be nullable (although I don't know why that'd ever be the case. Even tests should have a mock/test instance created)

Reply:
Tests pass undefined.. :(